### ISSUE TITLE
Lock Tabs option for the Primary Tab Bar to prevent accidental dragging and hiding (Main Window)

### ISSUE THREAD

**Issue by slook** [nicotine-plus/nicotine-plus#1504]
### Describe the solution you'd like
An option to 'Lock Main Window Tabs', either in the Modes menu, Preferences or in the Primary Tab bar context-menu(s).

Locking the Tabs prevents drag n drop, and also disables the right-click context menu.

### Additional context
It is too easy to reorder tabs by accident, it is frustrating when something important has moved or disappeared.

New users may think that the Tabs are dynamic, for example they may think the 'Search Files' tab is a per-search item that re-spawns after doing a new search, and they may decide to hide it thinking that means like close search. Perhaps there shouldn't really be a Hide option for certain tabs, but as there is, it would be nice to have an option to lock this.

It is possible to brick the interface if something like 'Search Files', 'Downloads', 'User Browse' is hidden. It isn't immediately obvious how to re-enable them. Perhaps main window tabs should be locked by default, until the user has learned how to use the program... the proposed 'Lock Tabs' option could be buried away in the Modes menu or in the Preferences.

**Comment by mathiascode** [comment]
> It is too easy to reorder tabs by accident, it is frustrating when something important has moved or disappeared.

Just curious, how does this happen, and can it be mitigated somehow? There used to be an option to disable reordering of tabs, but nobody I asked seemed to care about it.

**Comment by mathiascode** [comment]
> Perhaps there shouldn't really be a Hide option for certain tabs

People use this feature, so this isn't an option, but perhaps this functionality could be implemented as a list of tabs + checkboxes in the preferences instead?

**Comment by slook** [comment]
Yes the menu can sometimes be accidentally activated when using a device with a trackpad for example, then because the Hide menuitem is so close to the activation point then it is easily clicked.

It wouldn't be so bad if there was more than item in the context-menu. Then at least the mouse cursor would be far enough away from the offending menuitem that it wouldn't be so much of a problem. Perhaps a spacer can be added or some other menu item(s), that's what gave me the idea of the Lock Tabs button (which would be a global setting).

Options in the Preferences would be a much more appropriate option, because this is not something that needs configuring on a daily basis.
